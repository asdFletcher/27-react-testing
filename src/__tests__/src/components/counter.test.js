
const util = require('util');

import React from "react";

import renderer from "react-test-renderer";

import Counter from "../../../components/counter/counter.js";

describe("<counter />", () => {
  it("is alive", () => {
    expect(true).toBeTruthy();
  });

  it("correctly renders", () => {
    let component = mount(<Counter />);
    expect(component.find(".count").exists()).toBeTruthy();
    expect(component.find(".up").exists()).toBeTruthy();
    expect(component.find(".down").exists()).toBeTruthy();
  });

  it("changes state on a click", () => {
    let component = mount(<Counter />);
    let button = component.find(".down");
    
    expect(component.state("count")).toBe(0);
    button.simulate("click");
    expect(component.state("count")).toBe(-1);
    expect(component.find(".count").text()).toContain("-1");
  });

  it("changes state on a click", () => {
    let component = mount(<Counter />);
    let button = component.find(".up");
    
    expect(component.state("count")).toBe(0);
    button.simulate("click");
    expect(component.state("count")).toBe(1);
    expect(component.find(".count").text()).toContain("1");
  });

  it("renders correctly", () => {
    const tree = renderer.create(<Counter />).toJSON();
    expect(tree).toMatchSnapshot();
  });

});
