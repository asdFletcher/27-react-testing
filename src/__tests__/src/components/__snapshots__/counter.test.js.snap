// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<counter /> renders correctly 1`] = `
<section
  className="counter"
>
  <a
    className="down clicker"
    href="#"
    onClick={[Function]}
  >
    <svg
      height="1792"
      style={
        Object {
          "fill": "green",
        }
      }
      viewBox="0 0 1792 1792"
      width="1792"
    >
      <path
        d="M1600 736v192q0 40-28 68t-68 28h-416v416q0 40-28 68t-68 28h-192q-40 0-68-28t-28-68v-416h-416q-40 0-68-28t-28-68v-192q0-40 28-68t68-28h416v-416q0-40 28-68t68-28h192q40 0 68 28t28 68v416h416q40 0 68 28t28 68z"
      />
    </svg>
  </a>
  <span
    className="count "
  >
    0
  </span>
  <a
    className="up clicker"
    href="#"
    onClick={[Function]}
  >
    <svg
      height="1792"
      style={
        Object {
          "fill": "red",
        }
      }
      viewBox="0 0 1792 1792"
      width="1792"
    >
      <path
        d="M1600 736v192q0 40-28 68t-68 28h-1216q-40 0-68-28t-28-68v-192q0-40 28-68t68-28h1216q40 0 68 28t28 68z"
      />
    </svg>
  </a>
</section>
`;
